### PR TITLE
DOC: Add minimal geocoding example for docs

### PR DESCRIPTION
**Pull request by jdmcbr**
Shows the location of borough centers accessed through Google geocoding
API, along with the boundaries in the shapefile included in geopandas.

### CODE REVIEW

File: doc/source/geocoding.rst [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+locations of boroughs in New York City, and plots those locations along
+with the detailed borough boundary file included within ``geopandas``.
+
+.. _geopy: http://geopy.readthedocs.io/
+
+.. ipython:: python
+
+    boros = gpd.read_file(gpd.datasets.get_path("nybb"))
+    print(boros.BoroName)
+    boro_locations = gpd.tools.geocode(boros.BoroName, provider="google")
```
Comment (jorisvandenbossche) [inline doc/source/geocoding.rst]: I would also here just show how `boro_locations` looks like

File: doc/source/geocoding.rst [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
+
+    import matplotlib.pyplot as plt
+    fig, ax = plt.subplots()
 
+    boros.to_crs({"init": "epsg:4326"}).plot(ax=ax, color="white", edgecolor="black");
+    boro_locations.plot(ax=ax, color="red");
+    @savefig boro_centers_over_bounds.png
+    plt.show();
```
Comment (jorisvandenbossche) [inline doc/source/geocoding.rst]: The plt.show is not needed I think, you can put the `@savefig` above the previous plotting line

File: doc/source/geocoding.rst [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
-.. function:: geopandas.tools.geocode(strings, provider='googlev3', **kwargs)
 
-  Geocode a list of strings and return a GeoDataFrame containing the
-  resulting points in its ``geometry`` column.  Available
-  ``provider``s include ``googlev3``, ``bing``, ``google``, ``yahoo``,
-  ``mapquest``, and ``openmapquest``.  ``**kwargs`` will be passed as
-  parameters to the appropriate geocoder.
+Available ``provider`` arguments include ``'google'``, ``'bing'``,
+``'yahoo'``, and ``'openmapquest'``. See
+``geopy.geocoders.SERVICE_TO_GEOCODER`` for the full list.
```
Comment (jorisvandenbossche) [inline doc/source/geocoding.rst]: Maybe also mention you can also pass a geopy instance?

**Comment by jdmcbr** [comment]
@jorisvandenbossche Thanks for the suggestions. I thought the savefig behavior worked that way, but must have put it in the wrong place the first time I tried.

**Comment by jorisvandenbossche** [comment]
Thanks! (I just removed some redundant 'print')

**Pull request merged**